server/src/test/java/io/spine/examples/chatspn/server/chat/ChatDeletionTest.java

```diff
@@ -0,0 +1,176 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.server.chat;
+
+import io.spine.examples.chatspn.chat.Chat;
+import io.spine.examples.chatspn.chat.command.DeleteChat;
+import io.spine.examples.chatspn.chat.event.ChatDeleted;
+import io.spine.examples.chatspn.chat.event.ChatDeletionFailed;
+import io.spine.examples.chatspn.chat.event.ChatMarkedAsDeleted;
+import io.spine.examples.chatspn.chat.rejection.DeletionRejections.ChatCannotBeMarkedAsDeleted;
+import io.spine.examples.chatspn.message.Message;
+import io.spine.examples.chatspn.server.ChatsContext;
+import io.spine.examples.chatspn.server.message.MessageAggregate;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.core.given.GivenUserId;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.chatspn.message.MessageRemovalIdentifiersConverter.chatId;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.chatCannotBeMarkedAsDeletedFrom;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.chatDeletedFrom;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.chatDeletionFailedFrom;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.chatMarkedAsDeletedFrom;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.deleteChatCommand;
+import static io.spine.examples.chatspn.server.chat.given.ChatDeletionTestEnv.sendMessageTo;
+import static io.spine.examples.chatspn.server.chat.given.ChatTestEnv.createGroupChatIn;
+import static io.spine.examples.chatspn.server.chat.given.ChatTestEnv.createPersonalChatIn;
+
+@DisplayName("`ChatDeletion` should")
+final class ChatDeletionTest extends ContextAwareTest {
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("emit the `ChatDeleted` and archive itself if the chat can be deleted")
```

Comment:
I am not sure here. Do we really say "marked as deleted" and just archive it?

Why not delete (`setDeleted(true)`) altogether?